### PR TITLE
Firefox support

### PR DESCRIPTION
**Pull request by salmanulfarzy**
Saw #54 and still making a PR to support Firefox since the changes are minor. 

- Dropped `:-webkit-any(a, span)` selector which was introduced with https://github.com/sindresorhus/hide-files-on-github/commit/298ab686f73c3b0d49e96ced23af770552ea58a7 and had incompatibility with Firefox. Firefox only supports `:-moz-any()` and [not the newer `:matches()`][1]  pseudo-class.
- ~~Firefox supports `chrome` namespace and it was used for storage~~
- ~~Storage changed from `sync` to `local` (`sync` wasn't working with testing, Anyway to fix this ?)~~

[1]: https://caniuse.com/#feat=css-matches-pseudo

Changes tested with Firefox 62 and Chromium 68. 

If maintainers don't want to support Firefox because of maintenance burden, I would be happy to Firefox support here or submit it as separate extension on AMO under different name. Really  wish this would be accepted here because separate extension will most likely lead to code syncing and update issues.

### CODE REVIEW

**Comment by rugk** [comment]
> (sync wasn't working with testing, Anyway to fix this ?)

In general it does work. Maybe you did notw ait long enough or such stuff... In general I can say, it works.

**Comment by bfred-it** [comment]
The selector can change, fine. `sync` cannot; this breaks the option syncing in Chrome. If it doesn't work open an issue in bugzilla.

File: extension/api.js [hunk 1]
```diff
@@ -30,9 +30,9 @@ HideFilesOnGitHub.defaults = {
 
 HideFilesOnGitHub.storage = {
 	get: () => new Promise(resolve => {
-		chrome.storage.sync.get(HideFilesOnGitHub.defaults, resolve);
+		chrome.storage.local.get(HideFilesOnGitHub.defaults, resolve);
 	}),
 	set: obj => {
-		chrome.storage.sync.set(obj);
+		chrome.storage.local.set(obj);
```
[the remark this answers is not in the document]
Comment (salmanulfarzy) [reply extension/api.js]: Reverted `local` storage.

**Comment by salmanulfarzy** [comment]
`storage.sync` [requires add-on id][1] on `manifest.json`, Will updated with that.

[1]: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/WebExtensions_and_the_Add-on_ID#When_do_you_need_an_add-on_ID

**Comment by bfred-it** [comment]
Excellent! Thanks for finding that info.

I think Sindre's main gripe with Firefox is AMO's strict reviewers, so as long as we don't have to publish it to AMO and don't have to make sacrifices for it we can "support" it.

@sindresorhus what do you say?

**Comment by salmanulfarzy** [comment]
Can we setup auto publishing to AMO with travis similar to `refined-github` ?

**Comment by bfred-it** [comment]
> I think Sindre's main gripe with Firefox is AMO's strict reviewers, so **as long as we don't have to publish it to AMO** and don't have to make sacrifices for it we can "support" it.

**Comment by rugk** [comment]
You don't have to, if it is done automatically... 😉 

Anyway, I think reviews on AMO are quite fast and so on nowadays. (before ff v57 it was sometimes a lot more delayed) So as long as you abine by the rules (and don't include minified custom stuff without prividing corresponding source code), it should be all well.

**Comment by salmanulfarzy** [comment]
AMO review process has improved considerably after webext. I saw you guys encounter some deployment issues with RGH on AMO and solving it. This extension isn't complex as RGH so thought this won't be an issue and published it when this was merged, Excuse me for that. 

I'll remove the listing and move to new name ASAP. Hope you would reconsider publishing to Firefox.

**Comment by bfred-it** [comment]
Manual reviews still happen and developers still get bothered by reviewers and the extension can get blocked. Automatic deployment solves absolutely nothing. I don’t blame Sindre for not wanting to put up with it when you can just use Foxified to install from the Chrome Web Store.

**Comment by bfred-it** [comment]
Manual reviews still happen and developers still get bothered by reviewers and the extension can get blocked. Automatic deployment solves absolutely nothing. I don’t blame Sindre for not wanting to put up with it when you can just use Foxified to install from the Chrome Web Store.

**Comment by rugk** [comment]
I am glad that these reviews are done [as some recent events show](https://www.bleepingcomputer.com/news/security/mega-chrome-extension-hacked-to-steal-login-credentials-and-cryptocurrency/). Also, as a dev you can avoid it just by keeping to the rules. Once it is in, it is unlikely to get blocked unless you include malware or such stuff. So automatic deployment makes totally sense.

**Comment by matsest** [comment]
I saw the published add-on on Mozilla's site -- is it any reason why it's only for Firefox >= v62? Any way of easily adding it to v60?

**Pull request merged**